### ISSUE TITLE
Unexpected Panic - Internal Compiler Error

### ISSUE THREAD

**Issue by KenAKAFrosty** [rust-lang/rust#146334]
I'm afraid it's hard to replicate and I don't think a minimal repro is possible (it seems likely due to the fact that this is inherently not minimal). 

I fully understand this may limit how helpful this report is, and if the report needs to be ignored accordingly that's totally okay! 

I just wanted to at least provide the ice.txt file, and the compiler flags just in case it's still helpful

The generated ICE .txt file reads
```
thread 'rustc' panicked at /rustc-dev/1ed3cd7030718935a5c5e5c8f6581f36d8be179f/compiler/rustc_query_system/src/dep_graph/serialized.rs:284:21:
Error: A dep graph node (mir_drops_elaborated_and_const_checked) does not have an unique index. Running a clean build on a nightly compiler with `-Z incremental-verify-ich` can help narrow down the issue for reporting. A clean build may also work around the issue.

                     DepNode: DepKind { variant: 57 }(598f34c6118b72e9-2b30a3a648903ce)
stack backtrace:
   0:     0x75b11538ca75 - std::backtrace::Backtrace::create::h6e6334a01cac253b
   1:     0x75b11538c9c5 - std::backtrace::Backtrace::force_capture::h7d1e94eea457933c
   2:     0x75b1143e87b2 - std[f2f0971ac76601b7]::panicking::update_hook::<alloc[3c887bcc3df1f0a1]::boxed::Box<rustc_driver_impl[4b6d12a5e863dc6b]::install_ice_hook::{closure#1}>>::{closure#0}
   3:     0x75b11538d99f - std::panicking::panic_with_hook::h0d1ec96e086199a4
   4:     0x75b11538d75a - std::panicking::panic_handler::{{closure}}::he94a4ec25a0783eb
   5:     0x75b115387739 - std::sys::backtrace::__rust_end_short_backtrace::h9fdd742f7aa56a39
   6:     0x75b115367f6d - __rustc[110cd8a0e904276d]::rust_begin_unwind
   7:     0x75b111acedf0 - core::panicking::panic_fmt::h64aad7b1076844a2
   8:     0x75b116ea7a44 - rustc_incremental[81e724ebb47f8a09]::persist::load::setup_dep_graph
   9:     0x75b116c805be - rustc_interface[67817004344a6922]::interface::run_compiler::<(), rustc_driver_impl[4b6d12a5e863dc6b]::run_compiler::{closure#0}>::{closure#1}
  10:     0x75b1144026f4 - <rustc_thread_pool[bfc2ae2ac32cdf75]::job::StackJob<rustc_thread_pool[bfc2ae2ac32cdf75]::latch::LatchRef<rustc_thread_pool[bfc2ae2ac32cdf75]::latch::LockLatch>, <rustc_thread_pool[bfc2ae2ac32cdf75]::registry::Registry>::in_worker_cold<<rustc_thread_pool[bfc2ae2ac32cdf75]::thread_pool::ThreadPool>::install<rustc_interface[67817004344a6922]::util::run_in_thread_pool_with_globals<rustc_interface[67817004344a6922]::interface::run_compiler<(), rustc_driver_impl[4b6d12a5e863dc6b]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#5}::{closure#0}::{closure#1}::{closure#0}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()> as rustc_thread_pool[bfc2ae2ac32cdf75]::job::Job>::execute
  11:     0x75b1150b551f - <rustc_thread_pool[bfc2ae2ac32cdf75]::registry::WorkerThread>::wait_or_steal_until_cold
  12:     0x75b1150b3f01 - <rustc_thread_pool[bfc2ae2ac32cdf75]::registry::ThreadBuilder>::run
  13:     0x75b1143e09cd - std[f2f0971ac76601b7]::sys::backtrace::__rust_begin_short_backtrace::<<rustc_thread_pool[bfc2ae2ac32cdf75]::ThreadPoolBuilder>::build_scoped<rustc_interface[67817004344a6922]::util::run_in_thread_pool_with_globals<rustc_interface[67817004344a6922]::interface::run_compiler<(), rustc_driver_impl[4b6d12a5e863dc6b]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#5}::{closure#0}::{closure#0}, rustc_interface[67817004344a6922]::util::run_in_thread_pool_with_globals<rustc_interface[67817004344a6922]::interface::run_compiler<(), rustc_driver_impl[4b6d12a5e863dc6b]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#5}::{closure#0}::{closure#1}, ()>::{closure#0}::{closure#0}::{closure#0}, ()>
  14:     0x75b1143edb32 - <<std[f2f0971ac76601b7]::thread::Builder>::spawn_unchecked_<<rustc_thread_pool[bfc2ae2ac32cdf75]::ThreadPoolBuilder>::build_scoped<rustc_interface[67817004344a6922]::util::run_in_thread_pool_with_globals<rustc_interface[67817004344a6922]::interface::run_compiler<(), rustc_driver_impl[4b6d12a5e863dc6b]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#5}::{closure#0}::{closure#0}, rustc_interface[67817004344a6922]::util::run_in_thread_pool_with_globals<rustc_interface[67817004344a6922]::interface::run_compiler<(), rustc_driver_impl[4b6d12a5e863dc6b]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#5}::{closure#0}::{closure#1}, ()>::{closure#0}::{closure#0}::{closure#0}, ()>::{closure#1} as core[4c837b23521d86c9]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  15:     0x75b116bb440d - std::sys::pal::unix::thread::Thread::new::thread_start::h0a9a39ec7149870e
  16:     0x75b110494ac3 - start_thread
                               at ./nptl/pthread_create.c:442:8
  17:     0x75b110526850 - __GI___clone3
                               at ./misc/../sysdeps/unix/sysv/linux/x86_64/clone3.S:81:0
  18:                0x0 - <unknown>


rustc version: 1.91.0-nightly (1ed3cd703 2025-09-06)
platform: x86_64-unknown-linux-gnu
```

And the compiler flags are
```bash
note: compiler flags: --crate-type bin -C embed-bitcode=no -C debuginfo=2 -C incremental=[REDACTED] -Z threads=5
```

I can also at least confirm that running `cargo clean` and `cargo build` once more eliminated the issue

**Comment by jieyouxu** [comment]
Yeah, closing as duplicate, still thanks for reporting though since extra datapoints are very helpful.
